dtmi/osconfig/configuration-1.json

```diff
@@ -0,0 +1,109 @@
+{
+  "@context": "dtmi:dtdl:context;2",
+  "@id": "dtmi:osconfig:configuration;1",
+  "@type": "Interface",
+  "displayName": "Configuration",
+  "description": "Provides functionality to remotely manage OSConfig configuration on device",
+  "contents": [
+    {
+      "@type": "Property",
+      "name": "desiredConfiguration",
+      "schema": {
+        "@type": "Object",
+        "fields": [
+          {
+            "name": "refreshInterval",
+            "schema": "integer"
+          },
+          {
+            "name": "localManagementEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "fullLoggingEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "commandLoggingEnabled",
+            "schema": "boolean"
+          },
+          {
+            "name": "iotHubProtocol",
+            "schema": {
+              "@type": "Enum",
+              "valueSchema": "integer",
+              "enumValues": [
+                {
+                  "name": "auto",
+                  "enumValue": 0
+                },
+                {
+                  "name": "mqtt",
+                  "enumValue": 1
+                },
+                {
+                  "name": "mqttWebSocket",
+                  "enumValue": 2
+                }
+              ]
+            }
+          }
+        ]
+      },
+      "writable": true
+    },
+    {
+      "@type": "Property",
+      "name": "modelVersion",
+      "schema":"integer",
+      "writable": false
+    },
+    {
+      "@type": "Property",
+      "name": "refreshInterval",
+      "schema":"integer",
+      "writable": false
+    },
+    {
+      "@type": "Property",
+      "name": "localManagementEnabled",
+      "schema":"boolean",
+      "writable": false
+    },
+    {
+      "@type": "Property",
+      "name": "fullLoggingEnabled",
+      "schema":"boolean",
+      "writable": false
+    },
+    {
+      "@type": "Property",
+      "name": "commandLoggingEnabled",
+      "schema":"boolean",
+      "writable": false
+    },
+    {
+      "@type": "Property",
+      "name": "iotHubProtocol",
+      "schema": {
+        "@type": "Enum",
+        "valueSchema": "integer",
+        "enumValues": [
+          {
+            "name": "auto",
+            "enumValue": 0
+          },
+          { 
+            "name": "mqtt",
+            "enumValue": 1
```

Comment:
>"enumValue": 1 [](http://example.com/codeflow?start=12&length=14)

Should this DTMI property be different (enum names/values) than the reported MIM object?

Reply:
Current MIM for iotHubProtocol (reported): 

```JSON
"enumValues": [
    {
        "name": "mqtt",
        "enumValue": 0
    },
    {
        "name": "mqttWebSocket",
        "enumValue": 1
    }
]
```

Current DTMI for iotHubProtocol (reported):
```JSON
"enumValues": [
    {
        "name": "auto",
        "enumValue": 0
    },
    {
        "name": "mqtt",
        "enumValue": 1
    },
    {
        "name": "mqttWebSocket",
        "enumValue": 2
    }
]
```

For example: enumValue 0 means something different in MIM (**mqtt**) than in DTMI (**auto**). Is it accepted that this differs between MIM and DTMI?

Reply:
Yes, correct. This will be fixed, the MIM also needs to have same enums as in the PnP interface. Auto needs to be available for desired as well as it is an actual setting value in configuration JSON.